nx/lib/nx.ex

```diff
@@ -7390,13 +7429,165 @@ defmodule Nx do
   """
   @doc type: :ndim
   def sort(tensor, opts \\ []) do
-    opts = keyword!(opts, [axis: 0, comparator: :desc])
+    opts = keyword!(opts, axis: 0, comparator: :desc)
     comparator = opts[:comparator]
 
     %T{shape: shape, names: names} = tensor = to_tensor(tensor)
     axis = Nx.Shape.normalize_axis(shape, opts[:axis], names)
 
-    impl!(tensor).sort(tensor, tensor, axis: axis, comparator: comparator)
+    impl!(tensor).sort(tensor, tensor,
+      axis: axis,
+      comparator: comparator
+    )
+  end
+
+  @doc """
+  Sorts the tensor along the given axis with the
+  given comparator and returns the corresponding indices
+  of the original tensor in the new sorted positions.
+
+  If no axis is given, defaults to `0`.
+
+  ### Options
+
+  * `:axis`: The name or number of the corresponding axis on which the sort should be applied.
+  * `:comparator`: Can be `:asc`, `:desc` or a arity-2 function for comparison between two tensor elements. Defaults to `:asc`
+
+  ### Examples
+
+      iex> Nx.argsort(Nx.tensor([16, 23, 42, 4, 8, 15]))
+      #Nx.Tensor<
+        s64[6]
+        [3, 4, 5, 0, 1, 2]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :x)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 1],
+          [0, 1, 0]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :y)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 2],
+          [0, 2, 1]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :y, comparator: :asc)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [2, 0, 1],
+          [1, 2, 0]
+        ]
+      >
+
+      iex> tensor = Nx.tensor([[[4, 5], [2, 5], [5, 0]], [[1, 9], [2, 1], [2, 1]], [[0, -1], [-1, 0], [0, -1]], [[-1, 0], [0, -1], [-1, 0]]], names: [:x, :y, :z])
```

Comment:
This example is probably too big to follow as a doctest. Let's move it to `test/nx_test.exs` instead?

Reply:
Ok!